src/devices/Sgp30/Sgp30.cs

```diff
@@ -0,0 +1,279 @@
+                    // remaining initialisation measurements to complete.
+                    break;
+                }
+
+                Thread.Sleep(1000);
+            }
+
+            return result;
+        }
+
+        /// <summary>
+        /// Get the current eCO2 and TVOC measurements from the SGP30. NOTE: This should be called at least once per
+        /// second as per the device datasheet in order to maintain accuracy of the readings. If several seconds have
+        /// passed, call <see cref="InitialiseMeasurement" />.
+        /// </summary>
+        /// <returns>SGP30 measurement result.</returns>
+        public Sgp30Measurement GetMeasurement()
+        {
+            return GetData(SGP30_MEASURE_AIR_QUALITY);
+        }
+
+        /// <summary>
+        /// Get the raw signal data without on-device processing to TVOC ppb / eCO2 ppm values.
+        /// </summary>
+        /// <returns>Raw sensor data.</returns>
+        public Sgp30Measurement GetRawSignalData()
+        {
+            return GetData(SGP30_MEASURE_RAW_SIGNALS);
+        }
+
+        /// <summary>
+        /// Get device baseline measurement data.
+        /// </summary>
+        /// <returns>Measured baseline data.</returns>
+        public Sgp30Measurement GetBaseline()
+        {
+            return GetData(SGP30_GET_BASELINE);
+        }
+
+        /// <summary>
+        /// Generalised data read for getting Measurement, Raw Measurement and Baseline data.
+        /// </summary>
+        /// <param name="command">Command to send to the SGP30.</param>
+        /// <returns>Resulting data returned from SGP30.</returns>
+        private Sgp30Measurement GetData(ushort command)
+        {
+            Span<byte> resultBuffer = stackalloc byte[7];
+            _i2cDevice.Write(new ReadOnlySpan<byte>(new byte[] { (byte)((command & 0xFF00) >> 8), (byte)(command & 0x00FF) }));
+            Thread.Sleep(12);
+            _i2cDevice.Read(resultBuffer.Slice(1));
+            byte[] resultArray = resultBuffer.ToArray();
+
+            Sgp30Measurement result = new Sgp30Measurement
+            {
+                Eco2 = (ushort)(resultArray[1] << 8 | resultArray[2]),
+                Tvoc = (ushort)(resultArray[4] << 8 | resultArray[5])
+            };
+
+            byte[] checksums = new byte[]
+            {
+                CalculateChecksum(result.Eco2),
+                CalculateChecksum(result.Tvoc)
+            };
+
+            if (checksums[0] != resultArray[3] || checksums[1] != resultArray[6])
+            {
+                throw new ChecksumFailedException();
+            }
+
+            return result;
+        }
+
+        /// <summary>
+        /// Set SGP30 baseline values.
+        /// </summary>
+        /// <param name="tvoc">TVOC Baseline.</param>
+        /// <param name="eco2">eCO2 Baseline.</param>
+        public void SetBaseline(ushort tvoc, ushort eco2)
+        {
+            _i2cDevice.Write(new ReadOnlySpan<byte>(new byte[]
```

Comment:
You may want to use the BinaryPrimitives as well here and allocate a normal Span<byte> using stackalloc rather than a new byte array